pkg/generators/readers.go

```diff
@@ -237,6 +239,120 @@ func (g *ReadersGenerator) generateHelpers() error {
 			}
 			return
 		}
+
+		// ParseInteger reads a string and parses it to integer,
+		// if an error ocurred it returns a non-nil error.
+		func ParseInteger(query url.Values, parameterName string) (*int, error) {
+			values := query[parameterName]
+			count := len(values)
+			if count == 0 {
+				return nil, nil
+			}
+		   	if count > 1 {
+				err := fmt.Errorf(
+						"expected at most one value for parameter '%s' but got %d",
+						parameterName, count,
+				)
+				return nil, err
+			}
+			value := values[0]
+			parsedInt64, err := strconv.ParseInt(value, 10, 64)
+			if err != nil {
+				return nil, fmt.Errorf("value '%s' isn't valid for the '%s' parameter because it isn't an integer: %v",
+					value, parameterName, err)
```

Comment:
The style that I prefer to format error messages (and string formats in general) is to keep them in one line if the result is less than 100 characters. If it is longer then I like the `fmt.Errorf(` in one line, the format string in another line and the arguments in yet another line:

```go
err = fmt.Errorf(
        "value '%s' isn't valid ...",
        value, parameterName, err,
)
return nil, err
```

Can you use that style?